components/mediation-admin/org.wso2.carbon.mediation.templates/src/main/java/org/wso2/carbon/mediation/templates/services/TemplateEditorAdmin.java

```diff
@@ -337,7 +364,13 @@ public void saveTemplate(OMElement templateElement) throws AxisFault {
                     TemplateMediator templ = config.getSequenceTemplates().get(templateName);
                     if (templ != null) {
                         templ.init(getSynapseEnvironment());
-                        persistTemplate(templ);
+                        CAppArtifactDataService cAppArtifactDataService = ConfigHolder.getInstance().
+                                getcAppArtifactDataService();
+                        if (cAppArtifactDataService.isArtifactDeployedFromCApp(getTenantId(), ServiceBusConstants.TEMPLATE_TYPE + File.separator + templateName)) {
+                            cAppArtifactDataService.setEdited(getTenantId(), ServiceBusConstants.TEMPLATE_TYPE + File.separator + templateName);
```

Comment:
Use variable for 'ServiceBusConstants.TEMPLATE_TYPE + File.separator + templateName'